powerscale/helper/groupnet_helper.go

```diff
@@ -56,21 +56,30 @@ func UpdateGroupnetDataSourceState(ctx context.Context, groupnetState *models.Gr
 
 // UpdateGroupnetResourceState updates resource state.
 func UpdateGroupnetResourceState(ctx context.Context, groupnetModel *models.GroupnetModel, groupnetResponse *powerscale.V10NetworkGroupnetExtended) (err error) {
+	originModel := *groupnetModel
 
-	groupnetModel.ID = types.StringValue(*groupnetResponse.Id)
-	groupnetModel.Name = types.StringValue(*groupnetResponse.Name)
-	groupnetModel.AllowWildcardSubdomains = types.BoolValue(*groupnetResponse.AllowWildcardSubdomains)
-	groupnetModel.DNSCacheEnabled = types.BoolValue(*groupnetResponse.DnsCacheEnabled)
-	groupnetModel.ServerSideDNSSearch = types.BoolValue(*groupnetResponse.ServerSideDnsSearch)
+	if err = CopyFields(ctx, groupnetResponse, groupnetModel); err != nil {
+		return
+	}
```

Comment:
If CopyFields returns no error, the following code will not execute. Is that expected?
Cause I think it in most cases will not return error unless passing in parameters of wrong data type.

Reply:
@NavisJ thinking no matter errors, CopyFields should always be executed and the groupnetResponse is already parsed to groupnetModel.

Reply:
yes. You are right. I made a mistake with the function.